### ISSUE TITLE
deploy breaks wit PHP fatal error in 3.3.6

### ISSUE THREAD

**Issue by andreas-tritum** [TYPO3/Surf#727]
## Expected Behavior

Execution of surf deploy should not break in minor version update

## Actual Behavior

> PHP Fatal error:  TYPO3\Surf\Domain\Service\ShellCommandService and Psr\Log\LoggerAwareTrait define the same property ($logger) in the composition of TYPO3\Surf\Domain\Service\ShellCommandService. However, the definition differs and is considered incompatible. Class was composed in vendor/typo3/surf/src/Domain/Service/ShellCommandService.php on line 27

## Steps to Reproduce the Problem

composer init --require='typo3/surf:^3.3' --no-interaction
composer install
vendor/bin/surf deploy surfDeployment -vvv

## Specifications

  - Surf Version: 3.3.6
  - Application: 
  - PHP Version: 7.4

**Comment by chrosey** [comment]
same problem here

**Comment by simonschaufi** [comment]
sorry for breaking the latest release. Please downgrade for now and it will be fixed in the next release.

**Comment by YetiCGN** [comment]
Also this:

```
PHP Fatal error:  Uncaught Error: Call to a member function debug() on null in [...]/vendor/typo3/surf/src/Domain/Model/Workflow.php:46
```

**Comment by sabbelasichon** [comment]
Thanks. First of all, it would better to use the Phar-File, but we should of course try to fix it. Could you please show what version of psr/log you are using? I guess it is psr/log:^3.0

**Comment by andreas-tritum** [comment]
composer.lock says: `1.1.4`
```
{
    "name": "psr/log",
    "version": "1.1.4",
    "source": {
        "type": "git",
        "url": "https://github.com/php-fig/log.git",
        "reference": "d49695b909c3b7628b6289db5479a1c204601f11"
    },
    […]
}
```
```
composer why psr/log
monolog/monolog       1.27.1  requires  psr/log (~1.0)     
symfony/console       v5.4.21 conflicts psr/log (>=3)      
symfony/error-handler v5.4.21 requires  psr/log (^1|^2|^3) 
symfony/http-kernel   v5.4.21 requires  psr/log (^1|^2)
```

**Comment by andreas-tritum** [comment]
As current workaround I hope for next version excluding only 3.3.6:
```
"require": {
    "typo3/surf": "^3.3 !=3.3.6"
}
```

**Comment by simonschaufi** [comment]
One positive side effect of braking things is that we see that people still use surf and not everyone has switched to other tools like deployer or so. Thanks for using Surf! 🥳

**Comment by YetiCGN** [comment]
> One positive side effect of braking things is that we see that people still use surf and not everyone has switched to other tools like deployer or so. Thanks for using Surf! 🥳

Uhm... actually 3.3.5 and 3.3.6 accelerated our project to migrate to Deployer for TYPO3 projects as well.  😬 Sorry.

**Comment by simonschaufi** [comment]
@sabbelasichon you are more familiar with the self-update command. can you help?

**Comment by sabbelasichon** [comment]
@simonschaufi I have released version 3.3.8. The self-update command should be fixed there and also the logger problem. Thanks for your patience.
